utils.go

```diff
@@ -0,0 +1,43 @@
+package main
+
+import "time"
+
+func GetStartOfBudget(budget_type string, createdAt time.Time) time.Time {
```

Comment:
Seeing these again maybe it would be nice to suffix these methods to "OfCurrentBudgetCycle" instead of "OfBudget" to be clear it's based on the current time